doc/build.md

```diff
@@ -1,35 +1,31 @@
-# Build instructions for Linux
+# Build instructions
 
-See also [Istio Development](https://github.com/istio/istio/blob/master/devel/README.md) for setup and general tips.
-
-(For Windows and Mac, we recommend using a Linux virtual machine and/or [Vagrant-specific build instructions](build-vagrant.md) or [minikube build pod](minikube.md); Go code compiles on Mac but docker and proxy tests will fail on Mac)
 
-We are using [Bazel 0.5.2](https://github.com/bazelbuild/bazel/releases) as the main build system in Istio Pilot. The following command builds all targets in Istio Pilot:
 
-To avoid a `missing input file '//platform/kube:config'` error you need to `ln -s ~/.kube/config platform/kube/` or `make kubeconfig`, see [testing](testing.md) for details/why.
+## Environment
+We build on Linux. For Windows and Mac, we recommend using a Linux virtual machine and/or [Vagrant-specific build instructions](build-vagrant.md) or [minikube build pod](minikube.md). Go code compiles on Mac but docker and proxy tests will fail.
 
-    make build
+## Prerequisites
 
-Bazel uses `BUILD` files to keep track of dependencies between sources.  If you
-add a new source file or change the imports  please run the following command
-in the repository root to update all `BUILD` files:
+* [Setting up Go](https://github.com/istio/istio/blob/master/devel/README.md#setting-up-go)
+* [Setting up Bazel](https://github.com/istio/istio/blob/master/devel/README.md#setting-up-bazel)
 
-    make gazelle
-
-Data dependencies such as the ones used by tests require manual declaration in
-the `BUILD` files.
+## Initial Setup
 
-## Go tooling compatibility
+1. [Create a forked repository](https://github.com/istio/istio/blob/master/devel/README.md#fork-the-main-repository) in `$GOPATH/src/istio.io`.
+2. Run `make setup`. It well install the required tools and will [vendorize](https://golang.org/cmd/go/#hdr-Vendor_Directories) the dependencies.
```

Comment:
Please specify the exact version of Bazel. Project will not build with versions prior to 0.5.2.

Reply:
The magic involved in linking files and auto-vendorizing tends to be brittle and not guaranteed to work. I'm afraid by hiding both bazel and the brittle script, we're opening the doors to lots of questions about why the build fails.

Reply:
Added mentioning 0.5.2 version. So what are the sources of brittleness? What can be improved? If Bazel and the script are brittle, the questions will be asked anyway.

Please note that Makefile is already used in mixer and is mentioned in [Developing for Istio guide](https://github.com/istio/istio/tree/master/devel). This PR only tries to improve the existing instructions, which used Makefile before.

Reply:
Regarding dual/triple build systems - it seems this is the situation in mixer as well, and also in Kubernetes itself.
You have:
1. Bazel BUILD files
2. Scripts
3. Makefile

Makefile seems to be used to run scripts and to invoke Bazel. I do not know why this situation happened. My guess is that different tools are better at different tasks.